addons/l10n_account_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,249 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+from collections import defaultdict
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import UserError, RedirectWarning
+from odoo.osv import expression
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    # ------------------
+    # Fields declaration
+    # ------------------
+
+    l10n_account_withholding_display_withholding = fields.Boolean(compute='_compute_l10n_account_withholding_display_withholding')
+    l10n_account_withholding_withhold_tax = fields.Boolean(
+        string='Withhold Tax Amounts',
+        compute='_compute_from_lines',
+        store=True,
+        readonly=False,
+    )
+    l10n_account_withholding_line_ids = fields.One2many(
+        string="Withholding Lines",
+        comodel_name='account.payment.withholding.line',
+        inverse_name='payment_register_id',
+        compute='_compute_from_lines',
+        store=True,
+        readonly=False,
+    )
+    l10n_account_withholding_hide_number_col = fields.Boolean(compute='_compute_l10n_account_withholding_hide_number_col')
+    l10n_account_withholding_net_amount = fields.Monetary(
+        string='Net Amount',
+        help="Net amount after deducting the withholding lines",
+        compute='_compute_l10n_account_withholding_net_amount',
+    )
+    # We need to define the outstanding account of the payment in order for it to have the proper journal entry.
+    # To that end, we'll have this field required if we have a withholding tax impacting the payment and we don't have a payment account set on the payment method.
```

Comment:
This will be super useful: in Italy withholding is paid to Tax Agency or Pension Fund at a different time from the payment of the invoice.